### PR TITLE
CLOUDP-70064: mongocli ops-manager admin backup blockstore(s) create

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ CLOUDP-70064

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

## Further comments

```
./bin/mongocli om admin backup blockstore create --assignment --id test --uri mongodb://localhost:29022 -o json -P ops
{
  "id": "test",
  "uri": "mongodb://localhost:29022",
  "ssl": false,
  "assignmentEnabled": true,
  "encryptedCredentials": false,
  "loadFactor": 1
}
```

```
./bin/mongocli om admin backup blockstore create --assignment --id test --uri mongodb://localhost:29022  -P ops
Blockstore configuration 'test' created.
```

### CODE REVIEW

File: internal/cli/opsmanager/admin/backup/blockstore/create.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+}
+
+func (opts *CreateOpts) newBackupStore() *opsmngr.BackupStore {
+	backupStore := &opsmngr.BackupStore{
+		AdminBackupConfig: opsmngr.AdminBackupConfig{
+			ID:           opts.id,
+			URI:          opts.uri,
+			WriteConcern: opts.writeConcern,
+			Labels:       opts.label,
+		},
```
Comment (gssbzn) [inline internal/cli/opsmanager/admin/backup/blockstore/create.go:64]: [nit] feel free to do this in the future as we still have more create to go but let's maybe create a shared `* CreateOpts` with this logic that we can reuse with all this admin commands

File: internal/store/blockstores.go [hunk 2]
```diff
@@ -54,3 +58,14 @@ func (s *Store) DescribeBlockstore(blockstoreID string) (*opsmngr.BackupStore, e
 		return nil, fmt.Errorf("unsupported service: %s", s.service)
 	}
 }
+
+// BlockstoreCreater encapsulates the logic to manage different cloud providers
+func (s *Store) BlockstoreCreater(blockstore *opsmngr.BackupStore) (*opsmngr.BackupStore, error) {
```
Comment (gssbzn) [inline internal/store/blockstores.go:63]:
```suggestion
func (s *Store) CreateBlockstore(blockstore *opsmngr.BackupStore) (*opsmngr.BackupStore, error) {
```

File: internal/store/blockstores.go [hunk 3]
```diff
@@ -33,6 +33,10 @@ type BlockstoresDescriber interface {
 	DescribeBlockstore(string) (*opsmngr.BackupStore, error)
 }
 
+type BlockstoresCreater interface {
```
Comment (gssbzn) [inline internal/store/blockstores.go:36]: [typo]
```suggestion
type BlockstoresCreator interface {
```

**Review by gssbzn — CHANGES_REQUESTED** [review]
small typos and one optional future improvement

File: internal/cli/opsmanager/admin/backup/blockstore/create.go (hunk 1 above)
Comment (andreaangiolillo) [reply internal/cli/opsmanager/admin/backup/blockstore/create.go:64]: I am going to implement this in the update command

**Review by gssbzn — APPROVED** [review]
LGTM

**Pull request merged**